src/lib/confmgt/unitparse.c

```diff
@@ -142,10 +143,26 @@ config_parse_units(const char *val, const unit_table_t *u, int *ok)
 
   for ( ;u->unit;++u) {
     if (!strcasecmp(u->unit, cp)) {
-      if (use_float)
-        v = (uint64_t)(u->multiplier * d);
-      else
-        v *= u->multiplier;
+      if (use_float) {
+        d = u->multiplier * d;
+
+        if ( d > INT64_MAX ) {
```

Comment:
Let's make sure we're doing a floating point comparison here, using:
```suggestion
        if ( d > (double)INT64_MAX ) {
```

Reply:
See my updated suggestion, where I do both the integer and floating point checks.

Reply:
Thank you! This helped learn a bit more about safety comparison.

I'll update with your suggestion when the tests are ok.